internal/pkg/manifest/interpolate.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+)
+
+const (
+	predefinedEnvVarKeyForAppName = "COPILOT_APPLICATION_NAME"
+	predefinedEnvVarKeyForEnvName = "COPILOT_ENVIRONMENT_NAME"
+)
+
+var (
+	// Modified from docker/compose
+	interpolatorRegexp = regexp.MustCompile(`\${([_a-zA-Z][_a-zA-Z0-9]*)}`)
+)
+
+type interpolator struct {
+	predefinedEnvVars map[string]string
+}
+
+type predefinedEnvVar struct {
+	appName string
+	envName string
+}
+
+func newInterpolator(envVars predefinedEnvVar) *interpolator {
```

Comment:
Can we delete the `predefinedEnvVar` type?
```suggestion
func newInterpolator(appName, envName string) *interpolator {
```